src/john.c

```diff
@@ -1651,11 +1651,10 @@ static void john_init(char *name, int argc, char **argv)
 #endif
 
 	status_init(NULL, 1);
-	if (argc < 2 ||
-            (argc == 2 &&
-             (!strcasecmp(argv[1], "--help") ||
-              !strcasecmp(argv[1], "-h") ||
-              !strcasecmp(argv[1], "-help"))))
+	if (argc == 2 &&
+	     (!strcasecmp(argv[1], "--help") ||
+	      !strcasecmp(argv[1], "-h") ||
+	      !strcasecmp(argv[1], "-help")))
 	{
 		john_register_all(); /* for printing by opt_init() */
```

Comment:
I just realized this call to `john_register_all()` is obsolete (actually ever since 2015 in 9e3bb759)! Just remove it.

Reply:
...and as a resulting of that, you should actually move this whole thing from `john.c` to `opt_init()`, which would make its `int show_usage` parameter obsolete as well.

Reply:
Cool. I removed all references to `john_register_all()`, and moved the conditionals from john_init to `opt_init()`

Reply:
> I removed all references to `john_register_all()`

Oh, that will not work and wasn't what I said. I meant that one *call* to it only.

Reply:
3f6b666 is just like I meant, just revert 921eb19

Reply:
I just did. Thanks :) Unfortunately I'm down with a cold today, but hope to go through the rest in the next 1-2 days.